rec:  error: ‘NODCACHEDIRNOD’ was not declared in this scope

- [x] This is not a support question, I have read [about opensource](https://www.powerdns.com/opensource.html) and will send support questions to the IRC channel, [Github Discussions](https://github.com/PowerDNS/pdns/discussions/) or the mailing list.
- [x] I have read and understood the ['out in the open' support policy](https://blog.powerdns.com/2016/01/18/open-source-support-out-in-the-open/)

<!-- Tell us what is issue is about -->
 - Program:  Recursor
 - Issue type: Bug report

### Short description

A build of latest master fails with this error:

```
settings/cxxsettings-generated.cc:118:112: error: ‘NODCACHEDIRNOD’ was not declared in this scope
```

This was likely introduced by #13588.

### Environment
<!-- Tell us about the environment -->
 - Operating system: ubuntu:22.04 in docker
 - Software version: master as of 13 Dec 2023 ([f8a7f19](https://github.com/PowerDNS/pdns/commit/f8a7f19a2e0c710ca2ec8da0cc5031bb248c55dc))
 - Software source: compiled yourself

### Steps to reproduce

```
./configure --with-lua=luajit LDFLAGS=-rdynamic
make
```

### Expected behaviour

Successful build.

### Actual behaviour

```
52.37 settings/cxxsettings-generated.cc: In function ‘void pdns::settings::rec::defineOldStyleSettings()’:
52.37 settings/cxxsettings-generated.cc:118:112: error: ‘NODCACHEDIRNOD’ was not declared in this scope
52.37   118 |   ::arg().set("new-domain-history-dir", "Persist new domain tracking data here to persist between restarts") = NODCACHEDIRNOD;
```

### Other information
<!-- if you already did more digging into the issue, please provide all the information you gathered -->

This was likely introduced by #13588.

Tag [rec-5.0.0-rc1](https://github.com/PowerDNS/pdns/releases/tag/rec-5.0.0-rc1) does build correctly.

--- a/pdns/recursordist/Makefile.am
+++ b/pdns/recursordist/Makefile.am
@@ -21,10 +21,8 @@ AM_CXXFLAGS = \
 	-DPKGLIBDIR=\"$(pkglibdir)\" \
 	-DLOCALSTATEDIR=\"$(socketdir)\"
 
-if NOD_ENABLED
 AM_CXXFLAGS += \
 	-DNODCACHEDIRNOD=\"$(nodcachedir)/nod\" -DNODCACHEDIRUDR=\"$(nodcachedir)/udr\"
-endif
 
 if FSTRM
 AM_CPPFLAGS += \

